api/v1alpha3/conversion_test.go

```diff
@@ -39,18 +42,35 @@ func TestFuzzyConversion(t *testing.T) {
 		Spoke:  &VSphereCluster{},
 	}))
 	t.Run("for VSphereMachine", utilconversion.FuzzTestFunc(utilconversion.FuzzTestFuncInput{
-		Scheme: scheme,
-		Hub:    &nextver.VSphereMachine{},
-		Spoke:  &VSphereMachine{},
+		Scheme:      scheme,
+		Hub:         &nextver.VSphereMachine{},
+		Spoke:       &VSphereMachine{},
+		FuzzerFuncs: []fuzzer.FuzzerFuncs{CustomAdditionalDiskFuzzFunc},
 	}))
 	t.Run("for VSphereMachineTemplate", utilconversion.FuzzTestFunc(utilconversion.FuzzTestFuncInput{
-		Scheme: scheme,
-		Hub:    &nextver.VSphereMachineTemplate{},
-		Spoke:  &VSphereMachineTemplate{},
+		Scheme:      scheme,
+		Hub:         &nextver.VSphereMachineTemplate{},
+		Spoke:       &VSphereMachineTemplate{},
+		FuzzerFuncs: []fuzzer.FuzzerFuncs{CustomAdditionalDiskFuzzFunc},
 	}))
 	t.Run("for VSphereVM", utilconversion.FuzzTestFunc(utilconversion.FuzzTestFuncInput{
-		Scheme: scheme,
-		Hub:    &nextver.VSphereVM{},
-		Spoke:  &VSphereVM{},
+		Scheme:      scheme,
+		Hub:         &nextver.VSphereVM{},
+		Spoke:       &VSphereVM{},
+		FuzzerFuncs: []fuzzer.FuzzerFuncs{CustomAdditionalDiskFuzzFunc},
 	}))
 }
+
+func CustomAdditionalDiskFuzzFunc(_ runtimeserializer.CodecFactory) []interface{} {
+	return []interface{}{
+		CustomAdditionalDiskFuzzer,
+	}
+}
+
+func CustomAdditionalDiskFuzzer(in *nextver.VirtualMachineCloneSpec, c fuzz.Continue) {
+	c.FuzzNoCustom(in)
+
+	// AdditionalDisks field was introduced in v1alpha4. Setting those fields to nil to avoid
+	// v1alpha3 --> v1alpha4 --> v1alpha3 round trip errors
+	in.AdditionalDisks = nil
```

Comment:
hmmm why are we make this `nil`, `v1a3 -> v1a4 -> v1a3` should work without needing this

Reply:
Yes you are right, @gab-satchi  showed me a way to preserve Hub data on down conversion. I have a patch I'm working on.